dai_cadcad/flip.py

```diff
@@ -0,0 +1,134 @@
+""" Flip Module
+    Class-based representation of the Flip smart contract
+    (contains only what is necessary for the simulation)
+
+"""
+from dai_cadcad.pymaker.numeric import Wad, Rad
+from dai_cadcad.util import require
+
+
+class Bid:
+    bid = Rad(0)
+    lot = Wad(0)
+    guy = ""
+    tic = ""  # Date
+    end = ""  # Date
+    usr = ""
+    gal = ""
+    tab = Rad(0)
+    id = ""
+
+    def __init__(self, bid, lot, guy, tic, end, usr, gal, tab, id):
```

Comment:
let's have the parameter `id` here be called `bid_id` for verboseness (you can keep the name of the instance variable itself as `id`)